### PR TITLE
Remove IA32 Darwin support

### PR DESCRIPTION
**Pull request by mshinwell**
This patch removes support for 32-bit Darwin (macOS, iOS, etc) targets on Intel hardware.  This enables various special cases to be removed in the i386 backend that will in turn simplify a future pull request regarding symbol handling.

The current version of iOS does not support execution of 32-bit binaries any more, as far as I know.  I believe the current version of macOS (Mojave) is the last one that will support 32-bit binaries.

### CODE REVIEW

**Comment by nojb** [comment]
`README.adoc` also needs to be updated.

**Comment by nojb** [comment]
There are a couple more mentions of `macosx` in `configure.ac` which I think can be cleaned up.

**Comment by mshinwell** [comment]
Ah yes, I missed those as I was only looking for "darwin" there.
For some reason `autogen` produced changes in `configure` apart from the ones expected, so I manually deleted those parts of the diff.

**Comment by shindere** [comment]
Mark Shinwell (2019/03/04 03:57 -0800):
> Ah yes, I missed those as I was only looking for "darwin" there.
> For some reason `autogen` produced changes in `configure` apart from
> the ones expected, so I manually deleted those parts of the diff.

Perhaps configure.ac is no longer in sync with configure already before
your change? Or you don't have the same version of autoconf than the one
used to generate configure previously?

Please, do not hesitate to Cc me on such PRs!

File: Changes [hunk 1]
```diff
@@ -3,6 +3,12 @@ Working version
 
 (Changes that can break existing programs are marked with a "*")
 
+### Code generation and optimizations:
+
+- GPR#2278: Remove native code generation support for 32-bit Intel Darwin
```
Comment (xavierleroy) [inline Changes]: "Darwin" is a code name from the early 2000's.  Just call it "MacOS".

File: configure [hunk 2]
```diff
@@ -13526,12 +13522,6 @@ case $host in #(
     arch=i386; system=beos ;; #(
   i[3456]86-*-cygwin) :
     arch=i386; system=cygwin ;; #(
-  i[3456]86-*-darwin*) :
```
Comment (xavierleroy) [inline configure:13529]: Given the `if $arch64` below, it looks like it is / was possible to have a 64-bit x86 MacOS system reported as `i[3456]86-*-darwin`.  Are we sure this is no longer the case?

**Review by xavierleroy — COMMENTED** [review]
Looks good to me.  One suggestion and one question below.

File: Changes (hunk 1 above)
Comment (mshinwell) [reply Changes]: I think Darwin should be mentioned as there have been non-Mac systems using that kernel.  I've updated the entry to mention macOS and iOS too.

File: configure (hunk 2 above)
Comment (mshinwell) [reply configure:13529]: I'm sure enough.  I did some investigation on the web and also looked at our `config.guess` file.  It seems to me that it is unlikely we would obtain a target triplet that didn't start `x86_64-` on any vaguely recent macOS system.  If anyone is trying to cross compile to iOS, they will presumably have to specify the target triplet manually in any case, and as far as I understand it the `x86_64-` one has been canonical for some time.

**Comment by mshinwell** [comment]
@xavierleroy OK to merge this?

File: asmcomp/i386/NOTES.md [hunk 3]
```diff
@@ -18,5 +18,3 @@ Debian architecture name: `i386`
 * ELF application binary interface:
   _System V Application Binary Interface,
    Intel386 Architecture Processor Supplement_
-* MacOS X application binary interface:
-  _OS X ABI Function Call Guide: IA-32 Function Calling Conventions_
```
Comment (xavierleroy) [inline asmcomp/i386/NOTES.md:22]: Just for the record, could you please add the following to this `NOTES.md` file?  (I would have edited it myself but I don't have permissions, something that should not happen IMO.)
```
In OCaml versions 3.09.2 to 4.08, MacOS was supported by this port.  Support was removed in OCaml 4.09.
```

**Review by xavierleroy — COMMENTED** [review]
Looks good to me, with one "for the record" suggestion below.

File: asmcomp/i386/NOTES.md (hunk 3 above)
Comment (mshinwell) [reply asmcomp/i386/NOTES.md:22]: Done

**Review by xavierleroy — APPROVED** [review]
Thank you very much.  Good to go, merging!

**Pull request merged**